drivers/soundwire/stream.c

```diff
@@ -2051,6 +2060,39 @@ int sdw_stream_add_slave(struct sdw_slave *slave,
 			goto alloc_error;
 	}
 
+	if (!slave_prop->lane_control_support)
+		goto skip_lane_allocation;
+
+	for (i = 0; i < SDW_MAX_LANES; i++)
+		available_bandwidth[i] = bus->params.max_dr_freq - bus->params.bandwidth[i];
+
+	/* find available lane */
```

Comment:
find first available lane?

I think the algorithm described tries to use a minimal number of lanes, instead of spreading the load on multiple lanes? It could be interesting if we try to change the frequency as well :-)

Reply:
I don't know what's best in terms of power consumption.

There are two options:
- optimize the clock and reduce the number of lanes, i.e. only use additional lanes when the bandwidth is too low at max speed
- always use additional lanes to reduce the speed.

If the speed is fixed, it's not clear either. On one hand there's a need to enable a new line, but with the NRZI encoding if there's nothing to transmit then there are no transitions on the data line.